### PR TITLE
feat: set up codspeed

### PR DESCRIPTION
**Pull request by DaniPopes**
Adds [codspeed.io](https://codspeed.io) for benchmarking in CI.

Requires some minor tweaks to features to avoid "unknown feature" errors when building, and skipping some long benchmarks when running. Benchmarks run with valgrind and it struggles with a threads and huge maps.

[View online](https://codspeed.io/paradigmxyz/reth).

### CODE REVIEW

File: .github/scripts/codspeed-build.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+set -eo pipefail
+
+cmd=(cargo codspeed build --profile profiling)
+"${cmd[@]}" --features test-utils \
+    --workspace \
+    --exclude reth-libmdbx \
+    --exclude reth-transaction-pool
+
+# TODO: broken
+# "${cmd[@]}" -p reth-transaction-pool --features test-utils,arbitrary
```
Comment (mattsse) [inline .github/scripts/codspeed-build.sh:11]: me fixet

File: crates/transaction-pool/Cargo.toml [hunk 2]
```diff
@@ -76,45 +76,45 @@ serde_json.workspace = true
...
+    "reth-chainspec/arbitrary",
+    "reth-eth-wire-types/arbitrary",
+    "alloy-consensus/arbitrary",
+    "alloy-eips/arbitrary",
+    "alloy-primitives/arbitrary",
+    "bitflags/arbitrary",
+    "reth-primitives-traits/arbitrary",
+    "smallvec/arbitrary",
+    "revm-interpreter/arbitrary",
+    "revm-primitives/arbitrary",
```
Comment (mattsse) [inline crates/transaction-pool/Cargo.toml:117]: inb4 this is changed again

can't take this anymore 😭

**Review by mattsse — APPROVED** [review]

File: .github/scripts/codspeed-build.sh (hunk 1 above)
Comment (DaniPopes) [reply .github/scripts/codspeed-build.sh:11]: danke schon

File: crates/transaction-pool/Cargo.toml (hunk 2 above)
Comment (DaniPopes) [reply crates/transaction-pool/Cargo.toml:117]: it is what it is

**Review by onbjerg — COMMENTED** [review]
maybe we just ditch the other bench flows and just do codspeed?

File: .github/workflows/bench.yml
Comment (onbjerg) [inline .github/workflows/bench.yml:1]: i think remove all iai stuff cus it sucks anyway

**Review by onbjerg — APPROVED** [review]
fast

**Pull request merged**